.github/workflows/build-docker-image-and-binaries.yaml

```diff
@@ -48,27 +50,44 @@ jobs:
         run: |
           curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
 
-      - name: build binaries
-        env:
+      - name: build darwin binaries
+        env: 
           SEMVER: ${{ github.event.inputs.tag }}
+        if: matrix.os == 'macos-latest'
         run: |
-          if [ "$RUNNER_OS" == "Linux" ]
+          OS="darwin"
+          ARCH="${{ matrix.arch }}" 
+          if [ "$ARCH" == "arm64" ]
           then
-              OS="linux"
-              TOFND_PATH="/home/runner/work/tofnd/tofnd/target/release"
+            export HOMEBREW_NO_INSTALL_CLEANUP=TRUE
+            brew uninstall --ignore-dependencies gmp
+            rustup target add aarch64-apple-darwin
+            ARM_DEPENDENCY=$(brew fetch --force --bottle-tag=arm64_big_sur gmp | grep Downloaded | awk '{print $3}')
```

Comment:
I prefer to keep the workflow simple and move the case specific logic into a script rather have separate case dependent steps.

Reply:
sounds good. I will encapsulate the commands

Reply:
I added a script to encapsulate the complexity (install ARM64 dependency)